addons/knowledge/static/src/js/wysiwyg.js

```diff
@@ -0,0 +1,94 @@
+odoo.define('knowledge.wysiwyg', function (require) {
+'use strict';
+
+const core = require('web.core');
+const QWeb = core.qweb;
+
+const { DocumentWidget } = require('wysiwyg.widgets.media');
+const MediaDialog = require('wysiwyg.widgets.MediaDialog');
+const Link = require('wysiwyg.widgets.Link');
+const weWidgets = require('web_editor.widget');
+const Wysiwyg = require('web_editor.wysiwyg');
+
+const CustomDocumentWidget = DocumentWidget.extend({
```

Comment:
Impossible to review, I don't know the purpose of this widget and the code is really not straightforward.